ads/adhese.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function adhese(global, data) {
+  validateData(data, ['location','format', 'account', 'request_type']);
+  let adhTarget = '';
+  if (data['targeting'] != undefined) {
+    const t = data['targeting'];
+    for (const a in t) {
+      let s = a;
+      for (let x = 0; x < t[a].length; x++) {
+        s += t[a][x] + (t[a].length - 1 > x ? ';' : '');
```

Comment:
I would suggest using `forEach` here. Or can we store `t[a]` outside the loop. Better for review 🙏